src/inc/engine/game_renderer.h

```diff
@@ -3,12 +3,24 @@
 
 #include "cleanup.h"
 #include "game_component.h"
+#include "render_component.h"
+#include <type_traits>
 #include <vector>
 #include <SDL.h>
 #include <SDL_ttf.h>
 #include <stdio.h>
 #include <iostream>
 #include <SDL_image.h>
+#include "easy_sdl.h"
+
+template< typename, typename = std::void_t<>>
+struct IsRenderableT : std::false_type
+{};
+
+template< typename T>
+struct IsRenderableT< T, std::void_t< decltype( std::declval<T>().get_render_components())>>
```

Comment:
Type traits stuff here. Learned this technique from "C++ Templates: The Complete Guide, Second Edition" by Vandevoorde, Josuttis, and Gregor. The game renderer will be able to render anything that has a `get_render_components()` method that returns a list of `RenderComponent` implementations.